src/oci/conmon.rs

```diff
@@ -0,0 +1,292 @@
+    ExitDir(PathBuf),
+
+    /// Path to the program to execute when the container terminates its execution.
+    ExitCommand(PathBuf),
+
+    /// Delay before invoking the exit command (in seconds).
+    ExitDelay(Duration),
+
+    /// Additional arg to pass to the exit command. Can be specified multiple times.
+    ExitCommandArg(String),
+
+    /// Log file path.
+    LogPath(PathBuf),
+
+    /// Timeout in seconds.
+    Timeout(Duration),
+
+    /// Maximum size of log file.
+    LogSizeMax(u64),
+
+    /// Location of container attach sockets.
+    SocketDirPath(PathBuf),
+
+    /// Log to syslog (use with cgroupfs cgroup manager).
+    Syslog,
+
+    /// Print debug logs based on log level.
+    LogLevel(LevelFilter),
+
+    /// Additional tag to use for logging.
+    LogTag(String),
+
+    /// Do not manually call sync on logs after container shutdown.
+    NoSyncLog,
+
+    /// Allowing caller to keep the main conmon process as its child by only forking once.
+    Sync,
+}
+
+impl fmt::Display for Arg {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use crate::oci::conmon::Arg::*;
+        write!(f, "--")?;
+
+        fn write_kv<K, V>(f: &mut fmt::Formatter<'_>, key: K, value: V) -> fmt::Result
+        where
+            K: AsRef<str>,
+            V: fmt::Display,
+        {
+            write!(f, "{}={}", key.as_ref(), value)
+        }
+
+        match self {
+            Cid(id) => write_kv(f, self, id),
+            Cuuid(uuid) => write_kv(f, self, uuid),
+            Name(name) => write_kv(f, self, name),
+            Runtime(path) => write_kv(f, self, path.display()),
+            Restore(checkpoint) => write_kv(f, self, checkpoint),
+            RuntimeOpt(opt) => write_kv(f, self, opt),
+            RuntimeArg(opt) => write_kv(f, self, opt),
+            Bundle(path) => write_kv(f, self, path.display()),
+            PersistDir(path) => write_kv(f, self, path.display()),
+            ContainerPidfile(path) => write_kv(f, self, path.display()),
+            ConmonPidfile(path) => write_kv(f, self, path.display()),
+            ApiVersion(version) => write_kv(f, self, version),
+            ExecProcessSpec(path) => write_kv(f, self, path.display()),
+            ExitDir(path) => write_kv(f, self, path.display()),
+            ExitCommand(path) => write_kv(f, self, path.display()),
+            ExitDelay(duration) => write_kv(f, self, duration.as_secs()),
+            ExitCommandArg(arg) => write_kv(f, self, arg),
+            LogPath(path) => write_kv(f, self, path.display()),
+            Timeout(duration) => write_kv(f, self, duration.as_secs()),
+            LogSizeMax(size) => write_kv(f, self, size),
+            SocketDirPath(path) => write_kv(f, self, path.display()),
+            LogLevel(level) => write_kv(f, self, level),
+            LogTag(tag) => write_kv(f, self, tag),
+            _ => write!(f, "{}", self.as_ref()),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
this looks like a good set of tests to start with, might we want to add an e2e test for this? it seems like something that really requires e2e... just thinking out loud as I plan on writing some tests here soon.

Reply:
I agree this needs some e2e testing, I can follow up on that. :+1: